tools/upload-grid.mjs

```diff
@@ -0,0 +1,83 @@
+import fs from 'fs';
+import { getEnvKey } from './lib/envkeys.mjs';
+import { fetchProject, assignSessionsToSlotAndRoom } from './lib/project.mjs'
+import { validateSession } from './lib/validate.mjs';
+import seedrandom from 'seedrandom';
+
+function readconfig({ filename }) {
+   if (filename) {
+      let content = fs.readFileSync(filename).toString();
+      let regexp = /<pre id="data">[^<]+<\/pre>/s;
+      let data = content.match(regexp)[0].match(/\[\{.*\}\]/)[0];
```

Comment:
I'll have to dig into it a bit but I note that the two regular expressions could be combined, both to extract the data directly and to avoid relying on "<" as the delimiting parameter.